ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/participant_pruning_service.proto

```diff
@@ -36,6 +37,16 @@ service ParticipantPruningService {
   //   offset.
   rpc Prune (PruneRequest) returns (PruneResponse);
 
+  // Look up offset by timestamp to help determine a suitable pruning offset by ``record_time`` as exposed by the
+  // ``CommandCompletionService`` ``Checkpoint``s.
+  //
+  // Returns the last offset of the "first" event/transaction before or at the specified timestamp.
+  // Note that for ledger in which ``record_time`` increases monotonically with respect to offsets the resulting offset
+  // is uniquely defined (assuming events before or at the specified timestamp exist). For ledgers in which ``record_time``
+  // does not increase monotonically, there may be events with lower ``record_time`` at higher offsets than the returned
+  // offset. In such cases this command serves as an approximate heuristic.
+  rpc GetOffsetByTime (GetOffsetByTimeRequest) returns (GetOffsetByTimeResponse);
```

Comment:
Is there a reason why this should be limited to pruning? It seems like there might also be value in running something like "give me the transactions that happened yesterday" (and e.g. produce a daml dump based on that) where this could come in handy. That would suggest placing it closer to `GetLedgerEnd`.

Reply:
Yes, I was thinking about this too. In fact the completions/and or transactions service could then also be a good place for the `GetLedgerBegin` that @bame-da  was looking for.

Reply:
Good point @moritzkiefer-da . We would need though to make the return value specific to the transaction events that the `read_as` parties can see, as otherwise we leak the information of whether there are further events visible to parties other than the ones in `read_as`.

That in turn means that the pruning service lookup requires a different query, as it should consider all events stored in the IndexDB _independent_ of their per-party visibility.